kubeadm/hack/e2e-cluster-gcp.sh

```diff
@@ -0,0 +1,139 @@
+SECONDS=0
+while true; do
+	set +o errexit
+	timeout 30 gcloud compute ssh "root@${ctrlPlaneNodeName}" --command "KUBECONFIG=/etc/kubernetes/admin.conf kubectl version"
+	if [ $? -eq 0 ]; then
+		gcloud compute scp "root@${ctrlPlaneNodeName}:/etc/kubernetes/admin.conf" "$KUBECONFIG"
+		break
+	else
+		if [ $SECONDS -ge 600 ]; then
+			echo "Failed waiting for control plane to become ready"
+			exit 1
+		fi
+		sleep 10
+	fi
+done
+set -o errexit
+
+privateIP=$(echo $controlPlane | jq -r '.[] | .networkInterfaces[0].networkIP')
+externalIP=$(echo $controlPlane | jq -r '.[] | .networkInterfaces[0].accessConfigs[0].natIP')
+sed -i "s/${privateIP}/${externalIP}/g" $KUBECONFIG
+
+k8sVersion=$(kubectl version -ojson | jq -r .serverVersion.gitVersion)
+sed "s/VERSION/${k8sVersion}/" "$REPO_ROOT/kubeadm/hack/startup/windows.ps1" > $scratchDir/windows.ps1
+joinCmd="$(gcloud compute ssh root@${ctrlPlaneNodeName} --command "kubeadm token create --print-join-command") --ignore-preflight-errors=IsPrivilegedUser"
+
+for i in $(seq 1 $NODE_COUNT); do
+	gcloud compute instances create "$CLUSTER_NAME-windows-node-$i" \
+		--metadata-from-file windows-startup-script-ps1="$scratchDir/windows.ps1" \
+		--metadata join_cmd="$joinCmd" \
+		--image-project windows-cloud \
+		--network "$CLUSTER_NAME" \
+		--image-family windows-2019-core-for-containers \
+		--labels=cluster="$CLUSTER_NAME" \
+		--machine-type n1-standard-4
+done
+
+# build kubectl and e2e.test
+pushd $(go env GOPATH)/src/k8s.io/kubernetes
+git checkout "$k8sVersion"
+make all WHAT="test/e2e/e2e.test cmd/kubectl"
+export PATH="$PWD/_output/bin:$PATH"
+
+# install flannel overlay & kube-proxy
+curl -sL https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml | \
+	sed 's/"Type": "vxlan"/"Type": "vxlan", "VNI": 4096, "Port": 4789/' | kubectl apply -f-
+
+curl -sL https://github.com/kubernetes-sigs/sig-windows-tools/releases/latest/download/kube-proxy.yml | \
+	sed "s/VERSION/${k8sVersion}/g" | kubectl apply -f -
+
+kubectl apply -f https://github.com/kubernetes-sigs/sig-windows-tools/releases/latest/download/flannel-overlay.yml
+
+# wait for all the nodes to appear
+SECONDS=0
+while true; do
+	nodeCount=$(kubectl get nodes -l kubernetes.io/os=windows --no-headers | wc -l)
+	if [ $nodeCount -eq $NODE_COUNT ]; then
+		break
+	else
+		if [ $SECONDS -ge 600 ]; then
+			echo "Failed waiting for Windows nodes to join the cluster"
+			exit 1
+		fi
+		sleep 10
+	fi
+done
+
+# wait for all kube-system pods to become ready
+kubectl -n kube-system wait --for=condition=Ready pods --all --timeout=15m
+
+# setting this env prevents ginkgo e2e from trying to run provider setup
+export KUBERNETES_CONFORMANCE_TEST="y"
+
+$(go env GOPATH)/src/sigs.k8s.io/windows-testing/gce/run-e2e.sh \
+	--provider=skeleton --num-nodes=$(expr $NODE_COUNT + 1) \
+	--ginkgo.focus="$FOCUS" --ginkgo.skip="$SKIP" \
+	--node-os-distro=windows --report-dir="$ARTIFACTS" \
+	--disable-log-dump=true
```

Comment:
```
  # ginkgo regexes
  SKIP="${SKIP:-}"
  FOCUS="${FOCUS:-"\\[Conformance\\]"}"
  # if we set PARALLEL=true, skip serial tests set --ginkgo-parallel
  if [ "${PARALLEL:-false}" = "true" ]; then
    export GINKGO_PARALLEL=y
    if [ -z "${SKIP}" ]; then
      SKIP="\\[Serial\\]"
    else
      SKIP="\\[Serial\\]|${SKIP}"
    fi
  fi
```

Reply:
$NODE_COUNT should be N workers

Reply:
as mentioned on slack i think we should have something like that here above the `run-e2e.sh` call:
```
# split k8sVersion into major, minor
pushd $(go env GOPATH)/src/k8s.io/kubernetes/kubernetes
git fetch https://github.com/kubernetes/kubernetes.git release-$major.$minor
git checkout release-$major.$minor
popd
```

logs of what prow does:
https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-kubeadm-kinder-1-17/1237332976740601859/clone-log.txt